### PR TITLE
Update ppo.py

### PR DESCRIPTION
**Pull request by ZachariahRosenberg**
Moved the KL early stop to before the optimizer step. If KL is too great, we don't want to take the gradient step.

Also, you can consider removing the `kl = mpi_avg()` line. The line above it is sufficient.

### CODE REVIEW

**Comment by kashif** [comment]
@ZachariahRosenberg yes I think you are right, if we were checking if `kl > target_kl` then we could put the training inside that that check, but here we are checking `kl > 1.5 * target_kl` which checks if the current `kl` over-shoots the target by 1.5* ... then it stops (i see that it could have overshot it from one step to the next)....

regarding the `kl = mpi_avg(...)` one is the local mean and the second is the mean over the `n` workers which are running in parallel... and when `n=1` this function does nothing...

hope that answers your queries, and thanks again for looking into this.

**Comment by ZachariahRosenberg** [comment]
Thanks for reviewing my suggestions!

Got it re mpi_avg.

re KL, not sure how the 1.5* affects whether the check should come before or after the optimizer step. In the original case, we make the step and then check whether we should do that again. In the amended case, we make the check and then decide whether we want to take the step.

It isn't a big difference. As you mentioned worst case is a single step that overshoots. The larger change is perhaps in readability.

**Comment by kashif** [comment]
@ZachariahRosenberg yes you are right, the logger line would have to be outside the loop since it records the steps, and then you might as well check for `kl > target_kl` and break... I wanted to follow the original tensorflow flow but yes try out your way to see if it makes a difference.
